src/ripple/protocol/STObject.h

```diff
@@ -401,7 +401,7 @@ class STObject
         if (!cf)
             throw std::runtime_error ("Wrong field type");
 
-        cf->setValue (value);
+        cf->setValue (std::move (value));
```

Comment:
I think the `static_assert` is a good suggestion.  Maybe test `is_lvalue_reference` instead since moving from an rvalue reference would be fine too.